### ISSUE TITLE
Support for Plugin Directories

### ISSUE THREAD

**Issue by akhilman** [datafolklabs/cement#350]
For now cement checks <pluginname>.py file existence and do not allows to load plugins from directories like:
```
pluginname/
    __init__.py -- contains load(app)
    controller.py -- contains controller
    morecode.py -- contains more random code
```
plugin directories will be useful for large plugin modules with submodules. 

I can try to hack plugin handler to allow load plugins from directories.

**Comment by derks** [comment]
This is a great idea, and should not be hard to implement at all (just putting `load()` in __init__.py or similar like you suggested).

**Comment by akhilman** [comment]
Ok. I'll do this and will make pull request.

**Comment by akhilman** [comment]
Also I would like to have ability to load precomiled plugins. Like Cython does.

**Comment by derks** [comment]
@akhilman can you create a new issue regarding the precompiled Cython plugins? I'd like to handle that separately from this feature request.  Would like some more context around what you'd like to accomplish as well.  Thanks.

**Comment by akhilman** [comment]
@derks ok, I'll do as soon as I start to rewrite my application using cython.

**Comment by derks** [comment]
@akhilman this has been added to `master`, and will be included in `stable/2.10.0`.  Would you mind giving it a try and let me know how it works out for you?

**Comment by akhilman** [comment]
Directory plug-ins works well. Thank you.
